### PR TITLE
SDK-4148 internal package strategy

### PR DESCRIPTION
**Pull request by sergeyspryker**
https://spryker.atlassian.net/browse/SDK-4148

### CODE REVIEW

File: src/Upgrade/Application/Strategy/ReleaseApp/Processor/PackageManagerPackagesFetcher/InternalPackageManagerPackagesFetcher.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    protected function isPackageShouldBeUpdated(Package $package, array $requiredPackages): bool
+    {
+        return count(array_filter(
+            $requiredPackages,
+            static fn (Package $requiredPackage): bool => $requiredPackage->getName() === $package->getName()
+        )) === 0;
+    }
+
+    /**
+     * @return bool
```
Comment (vol4onok) [inline src/Upgrade/Application/Strategy/ReleaseApp/Processor/PackageManagerPackagesFetcher/InternalPackageManagerPackagesFetcher.php:60]: move all public functions above

File: tests/UpgradeTest/Application/Strategy/ReleaseApp/Processor/ModuleFetcherTest.php [hunk 2]
```diff
@@ -138,7 +167,7 @@ public function testRequireReturnsFailedResponseDtoIfItFailedToUpdateSubPackages
      */
     public function testRequireReturnsFailedResponseDtoIfRequireDevPackagesFailed(): void
     {
-        // Assert
+        // v
```
Comment (DmytroKlymanSpryker) [inline tests/UpgradeTest/Application/Strategy/ReleaseApp/Processor/ModuleFetcherTest.php:170]: Ctrl+V worked not as expected

File: src/Upgrade/Application/Strategy/ReleaseApp/Processor/ModuleFetcher.php [hunk 3]
```diff
@@ -67,30 +77,52 @@ public function require(ModuleDtoCollection $moduleCollection): PackageManagerRe
...
      * @param \Upgrade\Domain\Entity\Collection\PackageCollection $packageCollection
      *
+     * @throws \InvalidArgumentException
+     *
+     * @return \Upgrade\Application\Dto\PackageManagerPackagesDto
+     */
+    protected function getPackageManagerPackages(PackageCollection $packageCollection): PackageManagerPackagesDto
+    {
+        foreach ($this->packageManagerPackagesFetchers as $packageManagerPackagesFetcher) {
+            if (!$packageManagerPackagesFetcher->isApplicable()) {
```
Comment (vol4onok) [inline src/Upgrade/Application/Strategy/ReleaseApp/Processor/ModuleFetcher.php:95]: why not?
`if ($packageManagerPackagesFetcher->isApplicable()) {
    return $packageManagerPackagesFetcher->fetchPackages($packageCollection);
}`

File: src/Upgrade/Application/Strategy/ReleaseApp/Processor/PackageManagerPackagesFetcher/ExternalPackageManagerPackagesFetcher.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+     */
+    protected function isPackageShouldBeUpdated(Package $package, array $requiredPackages): bool
+    {
+        return false;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isApplicable(): bool
```
Comment (vol4onok) [inline src/Upgrade/Application/Strategy/ReleaseApp/Processor/PackageManagerPackagesFetcher/ExternalPackageManagerPackagesFetcher.php:50]: the same

**Review by vol4onok — APPROVED** [review]

**Review by DmytroKlymanSpryker — APPROVED** [review]

File: src/Upgrade/Application/Strategy/ReleaseApp/Processor/ModuleFetcher.php (hunk 3 above)
Comment (sergeyspryker) [reply src/Upgrade/Application/Strategy/ReleaseApp/Processor/ModuleFetcher.php:95]: For this case is no difference but in more complex should as described above

File: src/Upgrade/Application/Strategy/ReleaseApp/Processor/PackageManagerPackagesFetcher/AbstractPackageManagerPackagesFetcher.php [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    protected bool $isReleaseGroupIntegratorEnabled;
+
+    /**
+     * @param \Upgrade\Application\Adapter\PackageManagerAdapterInterface $packageManager
+     * @param bool $isReleaseGroupIntegratorEnabled
+     */
+    public function __construct(PackageManagerAdapterInterface $packageManager, bool $isReleaseGroupIntegratorEnabled = false)
+    {
+        $this->packageManager = $packageManager;
+        $this->isReleaseGroupIntegratorEnabled = $isReleaseGroupIntegratorEnabled;
```
Comment (VladislavStrelchenko) [inline src/Upgrade/Application/Strategy/ReleaseApp/Processor/PackageManagerPackagesFetcher/AbstractPackageManagerPackagesFetcher.php:36]: the property is unused

**Review by VladislavStrelchenko — APPROVED** [review]

**Pull request merged**